web/src/lib/utils/cast/gcast-destination.svelte.ts

```diff
@@ -0,0 +1,264 @@
+import { CastDestinationType, CastState, type CastDestination } from '$lib/managers/cast-manager.svelte';
+import 'chromecast-caf-sender';
+import { Duration } from 'luxon';
+
+const FRAMEWORK_LINK = 'https://www.gstatic.com/cv/js/sender/v1/cast_sender.js?loadCastFramework=1';
+
+enum SESSION_DISCOVERY_CAUSE {
+  LOAD_MEDIA,
+  ACTIVE_SESSION,
+}
+
+export class GCastDestination implements CastDestination {
+  type = CastDestinationType.GCAST;
+  isAvailable = $state(false);
+  isConnected = $state(false);
+  currentTime: number | null = $state(null);
+  duration: number | null = $state(null);
+  castState = $state(CastState.IDLE);
+  receiverName: string | null = $state(null);
```

Comment:
`$state` also accepts a generic, which (IMO) is cleaner, so like `currentTime = $state<number | null>(null);`